android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/RulesFactoryTest.kt

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.rulesengine
+
+import com.google.android.fhir.logicalId
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.slot
+import io.mockk.spyk
+import io.mockk.verify
+import java.util.Date
+import org.hl7.fhir.r4.model.Address
+import org.hl7.fhir.r4.model.ContactPoint
+import org.hl7.fhir.r4.model.Enumerations
+import org.hl7.fhir.r4.model.HumanName
+import org.hl7.fhir.r4.model.Meta
+import org.hl7.fhir.r4.model.Patient
+import org.hl7.fhir.r4.model.Resource
+import org.hl7.fhir.r4.model.StringType
+import org.jeasy.rules.api.Facts
+import org.jeasy.rules.api.Rules
+import org.jeasy.rules.core.DefaultRulesEngine
+import org.joda.time.LocalDate
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentMatchers.any
+import org.robolectric.util.ReflectionHelpers
+import org.smartregister.fhircore.engine.app.fakes.Faker
+import org.smartregister.fhircore.engine.configuration.ConfigurationRegistry
+import org.smartregister.fhircore.engine.domain.model.RuleConfig
+import org.smartregister.fhircore.engine.robolectric.RobolectricTest
+import org.smartregister.fhircore.engine.util.fhirpath.FhirPathDataExtractor
+
+class RulesFactoryTest : RobolectricTest() {
+
+  private lateinit var rulesEngine: DefaultRulesEngine
+  private lateinit var fhirPathDataExtractor: FhirPathDataExtractor
+  private lateinit var configurationRegistry: ConfigurationRegistry
+  private lateinit var rulesFactory: RulesFactory
+  private lateinit var rulesEngineService: RulesEngineService
+
+  @Before
+  fun setUp() {
+    configurationRegistry = Faker.buildTestConfigurationRegistry(mockk())
+    fhirPathDataExtractor = mockk(relaxed = true)
+    rulesEngine = mockk()
+    rulesEngineService = mockk()
+    rulesFactory =
+      spyk(
+        RulesFactory(
+          configurationRegistry = configurationRegistry,
+          rulesEngineService = rulesEngineService
+        )
+      )
+  }
+
+  @Test
+  fun `init() populates facts`() {
```

Comment:
Can be renamed as per the recommendation in the [Code-Testing- wiki](https://github.com/opensrp/fhircore/wiki/Code-Testing)